src/main/java/frc/trigon/robot/commands/factories/AutonomousCommands.java

```diff
@@ -0,0 +1,99 @@
+package frc.trigon.robot.commands.factories;
+
+import com.pathplanner.lib.commands.PathPlannerAuto;
+import com.pathplanner.lib.controllers.PPHolonomicDriveController;
+import edu.wpi.first.math.geometry.Pose2d;
+import edu.wpi.first.math.geometry.Rotation2d;
+import edu.wpi.first.wpilibj.DriverStation;
+import edu.wpi.first.wpilibj2.command.Command;
+import edu.wpi.first.wpilibj2.command.InstantCommand;
+import edu.wpi.first.wpilibj2.command.ParallelCommandGroup;
+import frc.trigon.robot.RobotContainer;
+import frc.trigon.robot.constants.CameraConstants;
+import frc.trigon.robot.constants.ShootingConstants;
+import frc.trigon.robot.misc.objectdetectioncamera.ObjectDetectionCamera;
+import frc.trigon.robot.subsystems.MotorSubsystem;
+import frc.trigon.robot.subsystems.intake.IntakeCommands;
+import frc.trigon.robot.subsystems.intake.IntakeConstants;
+import frc.trigon.robot.subsystems.pitcher.PitcherCommands;
+import frc.trigon.robot.subsystems.shooter.ShooterCommands;
+import org.trigon.utilities.mirrorable.MirrorablePose2d;
+
+import java.util.Optional;
+import java.util.function.Supplier;
+
+/**
+ * A class that contains commands that are used during the 15-second autonomous period at the start of each match.
+ */
+public class AutonomousCommands {
+    private static final ObjectDetectionCamera NOTE_DETECTION_CAMERA = CameraConstants.NOTE_DETECTION_CAMERA;
+
+    public static Command getResetPoseToAutoPoseCommand(Supplier<String> pathName) {
+        return new InstantCommand(
+                () -> {
+                    if (DriverStation.isEnabled())
+                        return;
+                    final Pose2d autoStartPose = PathPlannerAuto.getStaringPoseFromAutoFile(pathName.get());
+                    RobotContainer.POSE_ESTIMATOR.resetPose(new MirrorablePose2d(autoStartPose, true).get());
+                }
+        ).ignoringDisable(true);
+    }
+
+    public static Command getNoteCollectionCommand() {
+        return IntakeCommands.getSetTargetStateCommand(IntakeConstants.IntakeState.COLLECT).onlyIf(() -> !RobotContainer.INTAKE.hasNote());
+    }
+
+    public static Command getFeedNoteCommand() {
+        return new ParallelCommandGroup(
+                IntakeCommands.getSetTargetStateCommand(IntakeConstants.IntakeState.FEED_SHOOTING),
+                GeneralCommands.getVisualizeNoteShootingCommand().onlyIf(MotorSubsystem::isExtensiveLoggingEnabled)
+        ).until(() -> !RobotContainer.INTAKE.hasNote());
+    }
+
+    public static Command getAlignToNoteCommand() {
+        return new InstantCommand(
+                () -> {
+                    NOTE_DETECTION_CAMERA.startTrackingBestObject();
+                    overrideRotation(AutonomousCommands::calculateRotationOverride);
+                }
+        );
+    }
+
+    public static Command getStopAligningToNoteCommand() {
```

Comment:
This would mean that it would only stop aligning at the end of the paths where it's called. For the most part this is fine since most paths end after a collection, but if for some reason I'd want to have the path go on a bit longer it would still align. I think it's best the way it is.

Reply:
Alright